platform/security/sts/security-sts-propertyclaimshandler/src/test/java/org/codice/ddf/security/sts/claims/property/TestAttributeFileClaimsHandler.java

```diff
@@ -44,120 +44,315 @@
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(getPathForValidTestAttributesFile());
+    attributeFileClaimsHandler.init();
 
-  ClaimsParameters localhostClaimsParameters;
+    // when
+    final List<URI> supportedClaimTypes = attributeFileClaimsHandler.getSupportedClaimTypes();
 
-  ClaimsParameters regexClaimsParameters;
+    // then
+    assertThat(
+        supportedClaimTypes,
+        containsInAnyOrder(URI.create("test"), URI.create("test1"), URI.create("reg")));
+  }
 
-  ClaimsParameters unknownClaimsParameters;
+  @Test
+  public void testRetrieveClaimValuesUser1() throws IOException {
+    // given
+    System.setProperty(SystemBaseUrl.HOST, "user1");
 
-  File userFile;
+    final AttributeFileClaimsHandler attributeFileClaimsHandler = new AttributeFileClaimsHandler();
+    attributeFileClaimsHandler.setAttributeFileLocation(getPathForValidTestAttributesFile());
+    attributeFileClaimsHandler.init();
 
-  @Before
-  public void setup() throws IOException {
-    System.setProperty("ddf.home", "testdir");
-    InputStream resourceAsStream =
-        TestAttributeFileClaimsHandler.class.getResourceAsStream("/users.attributes");
-    userFile = temporaryFolder.newFile("users.attributes");
-    FileOutputStream userFileOs = new FileOutputStream(userFile);
-    IOUtils.copy(resourceAsStream, userFileOs);
+    final ClaimCollection claimCollection = getClaimCollectionForValidTestAttributesFile();
 
-    attributeFileClaimsHandler.setAttributeFileLocation(userFile.getAbsolutePath());
-    attributeFileClaimsHandler.init();
+    final ClaimsParameters user1ClaimsParameters = new ClaimsParameters();
+    final Principal principal = mock(Principal.class);
+    when(principal.getName()).thenReturn("user1");
+    user1ClaimsParameters.setPrincipal(principal);
 
-    claimCollection = new ClaimCollection();
-    Claim claim = new Claim();
-    claim.setClaimType(URI.create("test"));
-    claimCollection.add(claim);
-    Claim claim1 = new Claim();
-    claim1.setClaimType(URI.create("test1"));
-    claimCollection.add(claim1);
-    Claim claim2 = new Claim();
-    claim2.setClaimType(URI.create("reg"));
-    claimCollection.add(claim2);
+    // when
+    final ProcessedClaimCollection processedClaims =
+        attributeFileClaimsHandler.retrieveClaimValues(claimCollection, user1ClaimsParameters);
+
+    // then
+    assertThat(
+        processedClaims,
+        containsInAnyOrder(
+            allOf(
+                hasProperty("claimType", is(URI.create("test"))),
+                hasProperty("values", containsInAnyOrder("testValue"))),
+            allOf(
+                hasProperty("claimType", is(URI.create("test1"))),
+                hasProperty("values", containsInAnyOrder("testing1", "testing2", "testing3")))));
+  }
+
+  @Test
+  public void testRetrieveClaimValuesLocalhost() throws IOException {
```

Comment:
❓Is this testing that no claims are returned for a user that doesn't exist in the config file? If so, could rename the test method to make that clear? And what's the difference between this test and `testNoMatchRetrieveClaimValues`?

Reply:
`localhost` is included in the users.attributes file being tested here:
```json
{
    "user1" : {
        "test" : "testValue",
        "test1" : [ "testing1", "testing2", "testing3" ]
    },
    "localhost" : {

    },
    ".*host.*" : {
        "reg" : "ex"
    }
}
```
`testNoMatchRetrieveClaimValues` is tested a name that isn't in the users.attributes file. I updated the principal name in `testNoMatchRetrieveClaimValues` to clarify this. Let me know if there is something I can do to make this clearer.